plugins/inputs/logql/query.go

```diff
@@ -0,0 +1,187 @@
+package logql
+
+import (
+	"context"
+	"fmt"
+	"maps"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+)
+
+type query struct {
+	Name    string `toml:"name"`
+	Query   string `toml:"query"`
+	Sorting string `toml:"sorting"`
+	Limit   uint64 `toml:"limit"`
+
+	client *client
+	url    *url.URL
+	values *url.Values
+	log    telegraf.Logger
+}
+
+func (q *query) init(c *client, log telegraf.Logger) error {
+	if q.Query == "" {
+		return fmt.Errorf("'query' cannot be empty for %q", c.url)
+	}
+
+	switch q.Sorting {
+	case "", "forward", "backward":
+		// Do nothing, those are valid
+	default:
+		return fmt.Errorf("invalid sorting direction %q", q.Sorting)
+	}
+
+	// Prepare the query information from the URL and given parameters
+	u, err := url.Parse(c.url)
+	if err != nil {
+		return fmt.Errorf("parsing URL %q failed: %w", q.client.url, err)
```

Comment:
The code references q.client.url, but q.client is not assigned until line 46. If the URL parsing fails, accessing q.client.url will cause a nil pointer dereference panic.

This issue wasn’t caught earlier because all CI tests currently pass valid URLs, so the failing scenario was never triggered. The bug only occurs when url.Parse() encounters an invalid URL, and there are no existing tests that cover this case.

```suggestion
		return fmt.Errorf("parsing URL %q failed: %w", c.url, err)
```